spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulLifecycle.java

```diff
@@ -170,7 +173,7 @@ private void deregister(String serviceId) {
 
 	@Override
 	protected boolean isEnabled() {
-		return properties.isEnabled();
+		return lifecycleProperties.isEnabled();
```

Comment:
How about `properties.isEnabled() && lifecycleProperties.isEnabled()`? That way if `spring.cloud.consul.enabled=false` the lifecycle isn't started as well.